File: 卡码/KamaCoder/Python/test13.py
#13. 镂空三角形
import sys

def printSpace(n):
	for i in range(n):
		print(" ",end="")

def printLine(x,n):
	print(x,end="")
	for i in range(n-2):
		print(" ",end="")
	print(x,end="")

def main():
	for line in sys.stdin:
		if line[0]=="@":
			break
		x,n=line.split()
		n=int(n)
		for i in range(1,n+1):
			printSpace(n-i)
			if i==1 or i==n:
				for j in range(2*i-1):
					print(x,end="")
			else:
				printLine(x,2*i-1)
			print()
		print()

if __name__ == '__main__':
	main()

File: 卡码/KamaCoder/C++/test13.cpp
//13. 镂空三角形
#include <iostream>
using namespace std;

void printSpace(int n){
	for(int i=0;i<n;i++) printf(" ");
}

void printLine(char x,int n){
	printf("%c",x);
	for(int i=0;i<n-2;i++) printf(" ");
	printf("%c",x);
}

int main(){
	char x;
	int n;
	while(scanf("%c %d\n",&x,&n)!=EOF){
		if(x=='@') break;
		for(int i=1;i<=n;i++){
			printSpace(n-i);
			if(i==1||i==n)
				for(int j=1;j<=2*i-1;j++) printf("%c",x);
			else printLine(x,2*i-1);
			printf("\n");
		}
		printf("\n");
	}
	return 0;
}